### PR TITLE
feat(email component): add Email component

### LINKED ISSUE #17: Creating initial component
### Description:
1. Create the Email component

### Outcome:
1. Our first working initial component for the package

### PR DESCRIPTION
**Pull request by niloysikdar**
### This PR covers:
add the first working initial component, `Email` component for the package

resolves #17

Signed-off-by: Niloy Sikdar <niloysikdar30@gmail.com>

### CODE REVIEW

File: src/components/Email/index.tsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { FC, ReactNode } from 'react';
+
+interface EmailPropType {
+  children?: ReactNode;
+}
+
+export const Email: FC<EmailPropType> = (props) => {
```
Comment (agentmilindu) [inline src/components/Email/index.tsx:7]: `React.FC` already contains `children` isn't it? Anyways, I don't prefer using `React.FC` as now it's advice not to use it. Let's use the `EmailPropType` directly? 

```
const Email = ({ children }: EmailProps): JSX.Element => { /** */}
```

See https://stackoverflow.com/a/59988810/1190184

File: src/components/Email/index.tsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { FC, ReactNode } from 'react';
+
+interface EmailPropType {
```
Comment (agentmilindu) [inline src/components/Email/index.tsx:3]: I think it's fine to define it as just `EmailProps`. React Material UI also uses the same pattern. What do you think?

**Review by agentmilindu — CHANGES_REQUESTED** [review]

File: src/components/Email/index.tsx [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+  children?: ReactNode;
+}
+
+export const Email: FC<EmailPropType> = (props) => {
+  return <div>{props.children}</div>;
+};
```
[the remark this answers is not in the document]
Comment (niloysikdar) [reply src/components/Email/index.tsx:9]: Yeah, cool, doing this and will follow the same from the next PRs

File: src/components/Email/index.tsx (hunk 1 above)
Comment (niloysikdar) [reply src/components/Email/index.tsx:7]: For some reason, the latest react version doesn't have default `children` for `FC`

File: src/components/Email/index.tsx (hunk 2 above)
Comment (niloysikdar) [reply src/components/Email/index.tsx:3]: I just added `Type` with it cz it's kind of like mentioning the types for the props, that's why.
But that's fine, changing that to `EmailProps`

File: src/components/Email/index.tsx (hunk 1 above)
Comment (agentmilindu) [reply src/components/Email/index.tsx:7]: Do you have any references?

Comment (niloysikdar) [reply src/components/Email/index.tsx:7]: https://reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html#updates-to-typescript-definitions

@agentmilindu

File: src/components/Email/index.tsx [hunk 4]
```diff
@@ -1,9 +1,9 @@
...
+import { ReactNode } from 'react';
 
-interface EmailPropType {
+interface EmailProps {
   children?: ReactNode;
 }
 
-export const Email: FC<EmailPropType> = (props) => {
-  return <div>{props.children}</div>;
+export const Email = ({ children }: EmailProps) => {
```
Comment (agentmilindu) [inline src/components/Email/index.tsx:7]: Let’s define the return type of the component also. Refer to my example

Comment (niloysikdar) [reply src/components/Email/index.tsx]: done

**Pull request merged**